src/TodoList.js

```diff
@@ -1,41 +1,101 @@
 class TodoList {
-  constructor () {
+  constructor() {
     this.id = 0
     this.items = []
   }
 
-  create (str) {
+  create(str, name) {
     this.id++
-    const item = { id: this.id, text: str, status: 'incomplete' }
+    // const item = {
+    //   id: this.id,
+    //   text: str,
+    //   status: 'incomplete',
+    //   dateCreated: new Date().toLocaleDateString()
+    // }
+    // this.items.push(item)
+    const item = new TodoItem(this.id, name, str, 'incomplete')
     this.items.push(item)
     return item
   }
 
-  showAll () {
-    return this.items
+  showAll() {
+    if (this.items.length === 1) return this.items
+    return this.limitTextTo20Chars()
   }
 
-  setComplete (id) {
+  setComplete(id) {
     const item = this.findBy(id)
     item.status = 'complete'
     return item
   }
 
-  getByStatus (status) {
-    return this.items.filter(item => item.status === status)
+  getByStatus(status) {
+    return this.items.filter((item) => item.status === status)
   }
 
-  findBy (id) {
-    const item = this.items.find(item => item.id === id)
+  findBy(id) {
+    const item = this.items.find((item) => item.id === id)
     if (item === undefined) throw new Error('Item not found')
     return item
   }
 
-  deleteBy (id) {
+  deleteBy(id) {
     const item = this.findBy(id)
     const index = this.items.indexOf(item)
     return this.items.splice(index, 1)[0]
   }
+
+  limitTextTo20Chars() {
+    const limitedTextTodos = this.items.map((item) => {
+      if (item.text.length > 20) {
+        return { ...item, text: `${item.text.substring(0, 20)}...` }
+      } else {
+        return { ...item }
+      }
+    })
```

Comment:
Nice. You can remove the `else` here since there are only two things happening:

```js
    const limitedTextTodos = this.items.map((item) => {
      if (item.text.length > 20) {
        return { ...item, text: `${item.text.substring(0, 20)}...` }
      }
        return { ...item }
    })
```